src/mds/Server.cc

```diff
@@ -3057,6 +3057,13 @@ void Server::handle_client_getattr(MDRequestRef& mdr, bool is_lookup)
     }
   }
 
+  // non-auth MDS needs to sync latest rstat from auth MDS 
+  // if getattr request wants dir rstat, so try to acquire rdlock
+  // on nestlock to trigger the whole sync process
+  if (!ref->is_auth() && (mask & CEPH_STAT_RSTAT) 
+      && (issued & CEPH_CAP_XATTR_EXCL) == 0)
```

Comment:
don't check (issued & CEPH_CAP_XATTR_EXCL). vxattrs has no corresponding caps

Reply:
Besides, it's better to add optimizition similar to commit c822b3e2 "mds: optimize getattr file size"

Reply:
Ok, I will take a look.